### ISSUE TITLE
Add css-clean to list of supported minifiers

### ISSUE THREAD

**Issue by strarsis** [NMFR/optimize-css-assets-webpack-plugin#10]
The clean-css minifier now [also supports](https://github.com/jakubpawlowicz/clean-css/pull/944) the cssnano `process(css, opts)` API (merged soon). 
It would be nice if [clean-css](https://github.com/jakubpawlowicz/clean-css) could be added to the list of supported minifiers of this webpack plugin.

**Comment by NMFR** [comment]
I don't understand whats intended.

Do you wish to add [clean-css](https://github.com/jakubpawlowicz/clean-css) to the README of this repo has an alternative has a CSS processor?

**Comment by strarsis** [comment]
Yes.

**Comment by NMFR** [comment]
OK.

When this https://github.com/jakubpawlowicz/clean-css/pull/944 gets merged and included in a release i will test it (and if it works it shall be added to the README has an alternative).

Thanks.

**Comment by strarsis** [comment]
@NMFR: Merged.

**Comment by NMFR** [comment]
I'll wait until the changes are included in the next release.

**Comment by NMFR** [comment]
The change has yet to be release in a final version.

Going to close this issue.

If you notice that the change reaches a final version please reply in this issue and i will test it and add it to the README.

**Comment by MaxMilton** [comment]
Sorry to necro but it's relevant... :ghost: 

`clean-css` does have [some apparent support](https://github.com/jakubpawlowicz/clean-css/pull/944) for `optimize-css-assets-webpack-plugin` but I wasn't able to get it working at all, let alone with source maps.

Instead, I  put together something a bit custom to get it running correctly with source map support. Might be useful for someone in the future:

```javascript
const path = require('path');
const OptimizeCSSPlugin = require('optimize-css-assets-webpack-plugin');
const CleanCSS = require('clean-css');

...

  plugins: [
    // Compress extracted CSS. We are using this plugin so that possible
    // duplicated CSS from different components can be deduped.
    new OptimizeCSSPlugin({
      // use clean-css instead of the default cssnano
      cssProcessor: {
        process: (css, options) => new Promise((resolve, reject) => {
          const opts = Object.assign(options.map ? {
            sourceMap: true,
            sourceMapInlineSources: true,
          } : {}, options);

          opts.returnPromise = true;
          const filename = path.basename(opts.to);

          new CleanCSS(opts)
            .minify({
              [filename]: {
                styles: css,
                sourceMap: opts.map && opts.map.prev, // probably empty but here just in case
              }
            })
            .then((result) => {
              if (result.warnings) console.warn(result.warnings);

              resolve({
                css: options.map
                  ? result.styles + `\n/*# sourceMappingURL=${filename}.map */`
                  : result.styles,
                map: result.sourceMap,
              });
            })
            .catch(reject);
        }),
      },
      cssProcessorOptions: {
        // passing an object will enable source maps OR comment out to disable
        map: {},

        // the actual clean-css options
        level: {
          1: { all: true },
          2: { all: true },
        },
      },
    }),
  ]

...

```

**Comment by glen-84** [comment]
@NMFR,

> If you notice that the change reaches a final version please reply in this issue and i will test it and add it to the README.

It was released in [version 4.2](https://github.com/jakubpawlowicz/clean-css#whats-new-in-version-42).

It seems to work, but I'm not yet able to get source maps to work. The [minify method](https://github.com/jakubpawlowicz/clean-css#minify-method) returns the source map in `output.sourceMap`, does that need to be passed along [here](https://github.com/jakubpawlowicz/clean-css/blob/887ac0e2d1f10e16663edf0719153f18597824d5/lib/clean.js#L60) somehow?
